packages/framework-types/src/concepts/query.ts

```diff
@@ -0,0 +1,14 @@
+import { Class } from '../typelevel'
+import { PropertyMetadata } from '@boostercloud/metadata-booster'
+import { CommandAuthorizer } from './.'
+export type QueryInput = Record<string, any>
+export interface QueryInterface<TQuery = unknown, THandleResult = unknown> extends Class<TQuery> {
+  handle(query: TQuery): Promise<THandleResult>
+}
+
+export interface QueryMetadata<TCommand = unknown> {
+  readonly class: QueryInterface<TCommand>
+  readonly properties: Array<PropertyMetadata>
+  readonly methods: Array<PropertyMetadata>
+  readonly authorizer: CommandAuthorizer
+}
```

Comment:
As I said in a comment above, maybe we can leverage that a command and query are exaclty the same thing and reuse the types. Let me know what you think.

Reply:
BTW, you are using the CommandAuthorizer type, and it should be the QueryAuthorizer (again, if they are exactly the same, we could use the same type in both command and queries)

Reply:
You got me :) I don't have a strong opinion on this but for sure we should follow the same approach. Perhaps having a good name for both cases will help me.